packages/evm-connector/src/EvmWalletConnector.ts

```diff
@@ -0,0 +1,426 @@
+import { hexToBytes } from '@ethereumjs/util';
+// External libraries
+import {
+  type BytesLike,
+  arrayify,
+  hexlify,
+  splitSignature,
+} from '@ethersproject/bytes';
+import memoize from 'memoizee';
+
+import {
+  type AbiMap,
+  Address,
+  type Asset,
+  type ConnectorMetadata,
+  FuelConnector,
+  type InputValue,
+  type JsonAbi,
+  type Network,
+  Predicate,
+  Provider,
+  type TransactionRequestLike,
+  type Version,
+  getPredicateRoot,
+  transactionRequestify,
+} from 'fuels';
+
+import { WINDOW } from './constants';
+import type { EVMWalletConnectorConfig } from './types';
+import type { EIP1193Provider } from './utils/eip-1193';
+import { METAMASK_ICON } from './utils/metamask-icon';
+import { predicates } from './utils/predicateResources';
+
+export class EVMWalletConnector extends FuelConnector {
+  ethProvider: EIP1193Provider | null = null;
+  fuelProvider: Provider | null = null;
+  private predicate: { abi: JsonAbi; bytecode: Uint8Array };
+  private setupLock = false;
+  private _currentAccount: string | null = null;
+  private config: Required<EVMWalletConnectorConfig>;
+  private _ethereumEvents = 0;
+
+  // metadata placeholder
+  metadata: ConnectorMetadata = {
+    image: METAMASK_ICON,
+    install: {
+      action: 'Install',
+      description: 'Install a ethereum Wallet to connect to Fuel',
+      link: 'https://ethereum.org/en/wallets/find-wallet/',
+    },
+  };
+
+  constructor(config: EVMWalletConnectorConfig = {}) {
+    super();
+    this.name = 'EVM wallet connector';
+    this.predicate = predicates['verification-predicate'];
+    this.installed = true;
+    this.config = Object.assign(config, {
+      fuelProvider: 'https://beta-5.fuel.network/graphql',
```

Comment:
should we keep this hard-coded on this package?

Reply:
I have changed this to:
```
this.config = Object.assign(config, {
      fuelProvider: config.fuelProvider || BETA_5_URL,
      ethProvider: config.ethProvider || window.ethereum,
    });
```
The reason for the hardcoding was to allow the connector to be instantiated in the same way as the other connectors, with no params needed:
```
const fuel = new Fuel({
  connectors: [
    new FuelWalletConnector(),
    new EVMWalletConnector(),
  ],
});
```
In which case the `EVMWalletConnector` defaults to beta-5.
I'm open to suggestions on if this should be changed